gms/client/src/main/java/com/linkedin/glossary/client/GlossaryNodes.java

```diff
@@ -0,0 +1,156 @@
+package com.linkedin.glossary.client;
+
+import com.linkedin.glossary.GlossaryNode;
+import com.linkedin.glossary.GlossaryNodeKey;
+import com.linkedin.glossary.GlossaryNodesFindBySearchRequestBuilder;
+import com.linkedin.glossary.GlossaryNodesRequestBuilders;
+import com.linkedin.common.urn.GlossaryNodeUrn;
+import com.linkedin.data.template.StringArray;
+import com.linkedin.metadata.query.Filter;
+import com.linkedin.metadata.query.SortCriterion;
+import com.linkedin.metadata.restli.BaseSearchableClient;
+import com.linkedin.r2.RemoteInvocationException;
+import com.linkedin.restli.client.BatchGetEntityRequest;
+import com.linkedin.restli.client.Client;
+import com.linkedin.restli.client.GetAllRequest;
+import com.linkedin.restli.client.GetRequest;
+import com.linkedin.restli.common.CollectionResponse;
+import com.linkedin.restli.common.ComplexResourceKey;
+import com.linkedin.restli.common.EmptyRecord;
+
+import javax.annotation.Nonnull;
+import javax.annotation.Nullable;
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+import java.util.stream.Collectors;
+
+import static com.linkedin.metadata.dao.utils.QueryUtils.newFilter;
+
+public class GlossaryNodes extends BaseSearchableClient<GlossaryNode> {
+
+  private static final GlossaryNodesRequestBuilders BUSINESS_NODES_REQUEST_BUILDERS = new GlossaryNodesRequestBuilders();
+
+  public GlossaryNodes(@Nonnull Client restliClient) {
+    super(restliClient);
+  }
+
+  /**
+   * Gets {@link GlossaryNode} model of the business node.
+   *
+   * @param urn business node urn
+   * @return {@link GlossaryNode} model of the corp user
+   * @throws RemoteInvocationException
+   */
+  @Nonnull
+  public GlossaryNode get(@Nonnull GlossaryNodeUrn urn)
+      throws RemoteInvocationException {
+    GetRequest<GlossaryNode> getRequest = BUSINESS_NODES_REQUEST_BUILDERS.get()
+        .id(new ComplexResourceKey<>(toGlossaryNodeKey(urn), new EmptyRecord()))
+        .build();
+
+    return _client.sendRequest(getRequest).getResponse().getEntity();
+  }
+
+  /**
+   * Batch gets list of {@link GlossaryNode} models of the business nodes
+   *
+   * @param urns list of business node urn
+   * @return map of {@link GlossaryNode} models of the business nodes
+   * @throws RemoteInvocationException
+   */
+  @Nonnull
+  public Map<GlossaryNodeUrn, GlossaryNode> batchGet(@Nonnull Set<GlossaryNodeUrn> urns)
+      throws RemoteInvocationException {
+    BatchGetEntityRequest<ComplexResourceKey<GlossaryNodeKey, EmptyRecord>, GlossaryNode> batchGetRequest
+        = BUSINESS_NODES_REQUEST_BUILDERS.batchGet()
+        .ids(urns.stream().map(this::getKeyFromUrn).collect(Collectors.toSet()))
+        .build();
```

Comment:
there is a new batch get methodology created since this PR was started- please refer to usages of BatchGetUtils in other clients to see how you can use it! It minimizes boilerplate and removes the possibility you may overload GMS with a oversized batch get.

Reply:
Will do that